### PR TITLE
Add omega calculation and change LSM diag option to RTs

### LINKED ISSUE #2372: Add updated omega calculation pass to ufs-wm
## Description
A new feature was added in FV3 to calculate omega for non-hydrostatic runs (https://github.com/ufs-community/ufs-weather-model/pull/2327). Changes were made in global-workflow to implement these changes into their runs (https://github.com/NOAA-EMC/global-workflow/pull/2751) and this PR will add the new omega calculation for all regression tests that are non-hydrostatic.

## Solution
Change default_vars.sh to switch on the omega calculation for non-hydrostatic runs, change the diag_tables to allow for 'omga' output, and add a new field in the global namelist.

## Related to
Related to https://github.com/ufs-community/ufs-weather-model/pull/2327 for the initial implementation of the new omega calculation and https://github.com/NOAA-EMC/global-workflow/pull/2751 for the relevant global workflow PR.

### PR DESCRIPTION
**Pull request by dpsarmie**
## Commit Queue Requirements:

- [X] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [X] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [X] Commit 'test_changes.list' from previous step
---
## Description:
This PR adds the new omega calculation for RTs that are run in non-hydrostatic mode. Changes have been made to diag_tables and only the global_control.nml.IN. The original implementation of the omega calculation in FV3 can be found here: #2327 and the related global workflow PR that implements this is here: https://github.com/NOAA-EMC/global-workflow/pull/2751 .

IOPT_DIAG has been set to 2 for the remaining coupled test cases that had it set to 1. This should be the new default moving forward with the merge of #2354. Related GW issue is https://github.com/NOAA-EMC/global-workflow/issues/2786.

### Commit Message:

```
* UFSWM - Adds pass_full_omega_to_physics_in_non_hydrostatic_model parameter in the global_control.nml.IN namelist and
          sets it to true for non-hydrostatic runs. An additional field (omga) has been added to the diag_tables.
```

### Priority:

* Normal

### UFSWM:

* Closes #2372 

### Sub component Pull Requests:

* None

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* PR Updates/Changes Baselines.

### Input data Changes:

* None.

### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

File: tests/default_vars.sh
Comment (XiaqiongZhou-NOAA) [inline tests/default_vars.sh:1]: "if [[ "${HYDROSTATIC}" == .false. ]]; then
  export UPDATE_FULL_OMEGA=.true.
fi"
I do not think it need condition here. Default is false, only global runs is .true.

**Comment by XiaqiongZhou-NOAA** [comment]
Thank you very much for taking care of this
 In terms of the diag_table used for regional and nesting, you do not need add omga since it will keep original way - calculate omega in post. No baseline should be changed for all regional and HAFS's regression tests.

**Comment by dpsarmie** [comment]
> Thank you very much for taking care of this In terms of the diag_table used for regional and nesting, you do not need add omga since it will keep original way - calculate omega in post. No baseline should be changed for all regional and HAFS's regression tests.

Ok, thank you for taking a look at this. I will add those changes later today.

**Comment by dpsarmie** [comment]
I went ahead and redid the PR. The only changes now are to the global RTs and the regional RTs will remain as is.

**Review by BrianCurtis-NOAA — DISMISSED** [review]

**Review by junwang-noaa — DISMISSED** [review]

**Comment by dpsarmie** [comment]
More changes were rolled into this PR. A couple of the RTs were missing the LSM option that should be a part of #2354. Those have been added and tested. 

@HelinWei-NOAA thanks for pointing out the issue and let me know if anything else needs to be rolled in here.

I'll edit the overall PR text to reflect this soon.

**Comment by FernandoAndrade-NOAA** [comment]
Please go ahead and resync your branch, we'd like to get started with testing this PR today, thank you.

**Comment by FernandoAndrade-NOAA** [comment]
Please also commit the updated test_changes.list to verify which baselines need updates.

**Comment by dpsarmie** [comment]
> Please also commit the updated test_changes.list to verify which baselines need updates.

I had to regenerate the list, I'll ping you here once it's committed.

**Comment by dpsarmie** [comment]
@FernandoAndrade-NOAA Ok test_changes.list was added. Let me know if you need anything else.

**Comment by FernandoAndrade-NOAA** [comment]
@BrianCurtis-NOAA getting started with baseline creation for this PR.

**Comment by FernandoAndrade-NOAA** [comment]
If manually editing test_changes.list in the future, please be sure to verify the editor is using LF instead of CRLF, otherwise it'll insert hidden characters that will cause issues

**Comment by BrianCurtis-NOAA** [comment]
Before testing can start it will need a new bl-date

**Comment by FernandoAndrade-NOAA** [comment]
Adding a note that the following tests failed comparison on Gaea after successful baseline recreation:
`control_wam_debug intel`
`gnv1_nested intel`

**Comment by BrianCurtis-NOAA** [comment]
> Adding a note that the following tests failed comparison on Gaea after successful baseline recreation: `control_wam_debug intel` `gnv1_nested intel`

neither of those show up in the test_changes.list and the wam_debug one failed on WCOSS2. regenerate those baselines and try those two again. @dpsarmie can you check your old run logs to see if those tests passed?

**Comment by dpsarmie** [comment]
> > Adding a note that the following tests failed comparison on Gaea after successful baseline recreation: `control_wam_debug intel` `gnv1_nested intel`
> 
> neither of those show up in the test_changes.list and the wam_debug one failed on WCOSS2. regenerate those baselines and try those two again. @dpsarmie can you check your old run logs to see if those tests passed?

Just checked. The run that generated the test_changes.list was on Hercules and those tests are turned off for the Hercules platform. I'll run the tests on Hera and report back soon.

**Comment by BrianCurtis-NOAA** [comment]
> > > Adding a note that the following tests failed comparison on Gaea after successful baseline recreation: `control_wam_debug intel` `gnv1_nested intel`
> > 
> > 
> > neither of those show up in the test_changes.list and the wam_debug one failed on WCOSS2. regenerate those baselines and try those two again. @dpsarmie can you check your old run logs to see if those tests passed?
> 
> Just checked. The run that generated the test_changes.list was on Hercules and those tests are turned off for the Hercules platform. I'll run the tests on Hera and report back soon.

@jkbk2004 This is a good data point to emphasize that we should work towards all tests being run on one platform or devise a testing method where all tests are run between two machines prior to joining the commit queue.

**Comment by dpsarmie** [comment]
Ok, full suite has run on Hera and yes, `control_wam_debug intel` and `gnv1_nested intel` are showing as needing new baselines.

**Comment by zach1221** [comment]
Ok, testing is complete, we can proceed with final review.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**